pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/PartitionUpsertMetadataManager.java

```diff
@@ -84,6 +84,16 @@ public interface PartitionUpsertMetadataManager extends Closeable {
    */
   GenericRow updateRecord(GenericRow record, RecordInfo recordInfo);
 
+  /**
+   * Remove from the primary key index when the PK are expired if TTL is enabled.
+   */
+  void removeExpiredPrimaryKeys(Comparable expiredTimestamp);
+
+  /**
+   * Persist validDocIds snapshot when the validDocIds is stable if TTL is enabled.
+   */
+  void persistSnapshotForStableSegments(long expiredTimestamp);
+
```

Comment:
Why is the `expiredTimestamp` in `removeExpiredPrimaryKeys` a `Comparable` , whereas it is a `long` in `persistSnapshotForStableSegments` ?

Reply:
@navina The comparison column value is a comparable, in the primaryKey to recordLocation map.